src/views/Home/hooks/useGetTopFarmsByApr.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState, useEffect, useMemo } from 'react'
+import { ChainId } from '@pancakeswap/sdk'
+import { useFarms, usePriceCakeBusd } from 'state/farms/hooks'
+import { useAppDispatch } from 'state'
+import { fetchFarmsPublicDataAsync, nonArchivedFarms } from 'state/farms'
+import { getFarmApr } from 'utils/apr'
+import BigNumber from 'bignumber.js'
+import { orderBy } from 'lodash'
+import { FarmWithStakedValue } from 'views/Farms/components/FarmCard/FarmCard'
+import { Farm } from 'state/types'
+
+enum FetchStatus {
+  NOT_FETCHED = 'not-fetched',
+  FETCHING = 'fetching',
+  SUCCESS = 'success',
+  FAILED = 'failed',
+}
+
+const useGetTopFarmsByApr = (isIntersecting: boolean) => {
+  const dispatch = useAppDispatch()
+  const { data: farms } = useFarms()
+  const [fetchStatus, setFetchStatus] = useState(FetchStatus.NOT_FETCHED)
+  const [topFarms, setTopFarms] = useState<FarmWithStakedValue[]>([null, null, null, null, null])
+
+  const cakePriceBusdAsString = usePriceCakeBusd().toString()
+
+  const cakePriceBusd = useMemo(() => {
+    return new BigNumber(cakePriceBusdAsString)
+  }, [cakePriceBusdAsString])
```

Comment:
This already returns a `BigNumber`

Reply:
The memoisation is required to pass `cakePriceBusd` into a dependency array. The BN returned by `usePriceCakeBusd`'s reference changes on each render even when the value is the same, so it triggers a rerender loop if used in a dep array.

I'm not sure if there's a better way to achieve this though? I've had to implement this kind of workaround in a few places to prevent rerenders. 

- https://github.com/pancakeswap/pancake-frontend/blob/7c6c60215e1fb62844a401251d339d21a577c7f0/src/state/pools/hooks.ts#L85-L111
- https://github.com/pancakeswap/pancake-frontend/blob/6d22cd2b36fa9bcad8f5151fd391ff24c70b551b/src/state/lottery/helpers.ts#L147-L157
- https://github.com/pancakeswap/pancake-frontend/blob/d932319d2d1dbeba5d0580ee6fce0340ef498cc1/src/views/Home/components/WinSection/LotteryCardContent.tsx#L35-L37

Reply:
If that is the case lets just do it in the hook itself.

Reply:
Sure thing. Done 👍